lib/langchain/agent/chain_of_thought_agent/chain_of_thought_agent.rb

```diff
@@ -67,18 +73,24 @@ def run(question:)
           action_input = response.match(/Action Input: "?(.*)"?/)&.send(:[], -1)
 
           # Find the Tool and call `execute`` with action_input as the input
-          tool = tools.find { |tool| tool.tool_name == action.strip }
-          Langchain.logger.info("[#{self.class.name}]".red + ": Invoking \"#{tool.class}\" Tool with \"#{action_input}\"")
+          if (tool = tools.find { |tool| tool.tool_name == action.strip })
+            Langchain.logger.info("[#{self.class.name}]".red + ": Invoking \"#{tool.class}\" Tool with \"#{action_input}\"")
 
-          # Call `execute` with action_input as the input
-          result = tool.execute(input: action_input)
+            # Call `execute` with action_input as the input
+            result = tool.execute(input: action_input)
+          else
+            Langchain.logger.info("[#{self.class.name}]".red + ": Tool \"#{action}\" not found.")
+
+            result = "Tool \"#{action}\" not found. I need to try a different tool."
```

Comment:
Raised a question about this in the https://github.com/andreibondarev/langchainrb/pull/163. What do you think @bborn, @andreibondarev ?

Reply:
Agreed but this change should just be isolated to the other PR.